File: Domain/Entities/Message.cs
namespace Domain.Entities
{
    public class Message
    {
        public Guid Id { get; set; }
        public Guid? SenderId {  get; set; }
        public Guid? ReceiverId { get; set; }
        public string? Content { get; set; }
        public DateTime CreatedDate { get; set; }
        public bool IsRead { get; set; }
        public Message? ReplyMessage { get; set; }
        public Guid? ReplyMessageId { get; set; }
        public List<MessageAttachment>? Attachments { get; set; }
        public ApplicationUser? Sender { get; set; }
        public ApplicationUser? Receiver { get; set; }

    }
}

File: WebApi/client/src/app/api/models/message.ts
export interface Message {
  id?: string;
  senderId?: string | null;
  senderName?: string;
  receiverId?: string | null;
  content?: string | null;
  createdDate?: string;
  replyMessageId?: string;
  replyMessageContent?: string;
  replyMessageSenderName?: string;
  isRead?: boolean;
  attachments?: Attachments[];
}

export interface Attachments {
  path: string;
  type: string;
  name: string;
}
